multiaddr/multiaddr.py

```diff
@@ -130,6 +131,12 @@ def __init__(self, addr):
         else:
             raise TypeError("MultiAddr must be bytes, str or another MultiAddr instance")
 
+    @classmethod
+    def join(cls, *addrs):
```

Comment:
From https://github.com/multiformats/py-multiaddr/issues/47#issuecomment-491626516, the `multiaddrs` in `join` is an `Iterable` just like `str.join`, whereas we get variable-length param here. Would it be more intuitive to make it `Iterable` as `str.join`?

Reply:
I aimed to make both methods as similar as possible to the corresponding methods on `str`.
If you look at the usage with `Multiaddr.encapsulate`, I would think that doing `Multiaddr.join(a1, a2)` is easier to read and write then `Multiaddr.join([a1, a2])`.
For passing in a list of values one can always do `Multiaddr.join(*addr_list)`, just like with the `','.join(*parts)` often seen.

Reply:
Good point about the fact that `join(a1, a2)` is easier to read/write, when we have `Multiaddr.encapsulate`.